### PR TITLE
fix dump table name error and add some test for dump database

### LINKED ISSUE #6310: Failed to dump database: pq: relation "o_auth2_application" does not exist
- Gitea version: master (b2e9894988a8cb486f8838f4bf532401124802c4)
- Git version: 2.17.1
- Operating system: Ubuntu Bionic
- Database:
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log:

````
$ ./gitea dump
2019/03/12 07:52:27 Creating tmp work dir: /tmp/gitea-dump-808312899
2019/03/12 07:52:27 Packing dump files...
2019/03/12 07:52:27 Dumping local repositories.../home/git/gitea-repositories
2019/03/12 07:55:21 Dumping database...
2019/03/12 07:55:25 Failed to dump database: pq: relation "o_auth2_application" does not exist
````

## Description

This problem started occurring starting with commit e777c6bdc6f12f9152335f8bfd66b956aedc9957

### PR DESCRIPTION
**Pull request by lunny**
Will fix #6310 and replace #6385

### CODE REVIEW

**Comment by lunny** [comment]
Will add some dump integration tests after this merged.

File: models/models.go [hunk 1]
```diff
@@ -51,8 +51,9 @@ type Engine interface {
 }
 
 var (
-	x      *xorm.Engine
-	tables []interface{}
+	x                *xorm.Engine
+	supportedDatabse = []string{"mysql", "postgres", "mssql"}
```
Comment (jolheiser) [inline models/models.go]: Small typo nit, should be `supportedDatabase` or `supportedDatabases` and refactored as needed where used.

**Review by jolheiser — CHANGES_REQUESTED** [review]

File: models/models.go (hunk 1 above)
Comment (lunny) [reply models/models.go]: done. It's strange that misspell check failed.

**Review by jolheiser — APPROVED** [review]

**Review by sapk — APPROVED** [review]

**Review by axifive — APPROVED** [review]

**Comment by axifive** [comment]
but need to run fmt

**Comment by techknowlogick** [comment]
@lunny merged. Please backport 😄

**Pull request merged**